docs/reference/release-notes/v2.28/index.md

```diff
@@ -0,0 +1,153 @@
+- `FAST_REFRESH` - Use React Fast Refresh instead of the legacy react-hot-loader for instantaneous feedback in your development server. Recommended for versions of React >= 17.0.
+
+## Improved Fast Refresh integration
+
+Gatsby had some support for `react-refresh` (or Fast Refresh as it can be referred to) already. The PR for a [preliminary Fast Refresh integration](https://github.com/gatsbyjs/gatsby/pull/26664) added a custom overlay, better error recovery and in general a better communication between Gatsby's dev server and React Fast Refresh. This will give you quicker feedback and an overall improved development experience.
+
+To use it, add a flag to your `gatsby-config.js`:
+
+```js
+// In your gatsby-config.js
+module.exports = {
+  // your existing config
+  flags: {
+    FAST_REFRESH: true,
+  },
+}
+```
+
+Visit the [umbrella issue about Fast Refresh](https://github.com/gatsbyjs/gatsby/discussions/28390) to see preview images and give feedback.
+
+## Making `gatsby develop` faster
+
+We continue working on speeding up development server startup:
+
+### Experimental: Lazy images in develop
+
+As more and more images are added to a Gatsby site, the slower the local development experience oftentimes becomes.
+You spend time waiting for images to process, instead of you know, developing! No longer!
+This experimental version of `gatsby-plugin-sharp` only does image processing when the page gets requested.
+
+To use it, make sure you have `gatsby-plugin-sharp@^2.10.0` and add `LAZY_IMAGES` to flags in `gatsby-config.js`:
+
+```js
+// In your gatsby-config.js
+module.exports = {
+  // your existing config
+  flags: {
+    LAZY_IMAGES: true,
+  },
+}
+```
+
+[Details and discussion](https://github.com/gatsbyjs/gatsby/discussions/27603).
+
+## Image plugin helpers
+
+This release adds new utility functions to help plugin authors add support for the [new Gatsby image component](https://github.com/gatsbyjs/gatsby/discussions/27950). If you maintain a source plugin, or if you want to help with one, take a look at [the RFC](https://github.com/gatsbyjs/gatsby/discussions/28241).
+
+## Removed experimental lazy page bundling
+
+In gatsby@2.27.0 we added [Experimental: Lazy page bundling](../v2.27/index.md#experimental-lazy-page-bundling-in-development) mode for `gatsby develop` that would delay compiling page templates until it was needed. While preliminary tests were very promising, we discovered few showstoppers that degraded development experience. [We decided to end the experiment](https://github.com/gatsbyjs/gatsby/discussions/28137#discussioncomment-138998) for now and shift our efforts to [Less aggressive cache invalidation](#less-aggressive-cache-invalidation).
+
+## Notable bugfixes
+
+- fix: do not ignore source file changes during recompilation [#28237](https://github.com/gatsbyjs/gatsby/pull/28237)
+- fix: invalidate page data when static query is added [#28349](https://github.com/gatsbyjs/gatsby/pull/28349)
```

Comment:
I think those descriptions do not tell much unless you know the context. Maybe we should change those to something more human-friendly:

```suggestion
- fix: hot reloading hangs on multiple fast saves in develop [#28237](https://github.com/gatsbyjs/gatsby/pull/28237)
- fix error: `The result of this StaticQuery could not be fetched` when static query is added [#28349](https://github.com/gatsbyjs/gatsby/pull/28349)
```

Not insisting though, just my preference.

Reply:
Ah yeah. The reason I hesitated going this way because `The result of this StaticQuery could not be fetched` is not completely fixed - it is fixed in some cases but the most important (when user views the page they are editing) is not yet. Maybe better to omit this completely

Reply:
> This is not complete fix however. It doesn't exactly fix the problem when staticQueryHashes changed for a page user currently view in the browser ( note TODO comments I added in added e2e test scenario and dev-loader ), but it handles at least some cases.

From PR description - so it's partial fix (some cases are handled - reason why this went in as standalone)

Reply:
I used the suggestion (coulndn't apply, because other changes happened in meantime :( )

But I did use `parial fix:` prefix for the static query to note that this is not everything that is needed but might help - still not great but I have no more ideas how to mark that some cases are handled but not every case is yet